modules/extensions-xml-support/src/main/java/org/mule/runtime/extension/internal/processor/ModuleOperationMessageProcessor.java

```diff
@@ -0,0 +1,362 @@
+   */
+  private List<ParameterModel> getAllProperties(ExtensionModel extensionModel) {
+    List<ParameterModel> result = new ArrayList<>();
+    extensionModel.getConfigurationModel(MODULE_CONFIG_GLOBAL_ELEMENT_NAME).ifPresent(configurationModel -> {
+      result.addAll(configurationModel.getAllParameterModels());
+      configurationModel.getConnectionProviderModel(MODULE_CONNECTION_GLOBAL_ELEMENT_NAME)
+          .ifPresent(connectionProviderModel -> result.addAll(connectionProviderModel.getAllParameterModels()));
+    });
+    return result;
+  }
+
+  /**
+   * To properly feed the {@link ExpressionManager#evaluate(String, DataType, BindingContext, CoreEvent)} we need to store the
+   * {@link MetadataType} per parameter, so that the {@link DataType} can be generated.
+   *
+   * @param parameters list of parameters taken from the XML
+   * @param parameterModels collection of elements taken from the matching {@link ExtensionModel}
+   * @return a collection of parameters to be later consumed in {@link #getEvaluatedValue(CoreEvent, String, MetadataType)}
+   */
+  private Map<String, Pair<String, MetadataType>> parseParameters(Map<String, String> parameters,
+                                                                  List<ParameterModel> parameterModels) {
+    final Map<String, Pair<String, MetadataType>> result = new HashMap<>();
+
+    for (ParameterModel parameterModel : parameterModels) {
+      final String parameterName = parameterModel.getName();
+      if (parameters.containsKey(parameterName)) {
+        final String xmlValue = parameters.get(parameterName).trim();
+        result.put(parameterName, new Pair<>(xmlValue, parameterModel.getType()));
+      }
+    }
+    return result;
+  }
+
+  @Override
+  public CoreEvent process(final CoreEvent event) throws MuleException {
+    return processToApply(event, this);
+  }
+
+  /**
+   * Given an {@code event}, it will consume from it ONLY the defined properties and parameters that were set when initializing
+   * this class to provide scoping for the inner list of processors.
+   */
+  @Override
+  public Publisher<CoreEvent> apply(Publisher<CoreEvent> publisher) {
+    return from(publisher)
+        .doOnNext(event -> {
+          final DefaultFlowCallStack flowCallStack = (DefaultFlowCallStack) event.getFlowCallStack();
+
+          final ComponentIdentifier identifier = (ComponentIdentifier) getAnnotation(ANNOTATION_NAME);
+          if (identifier.getNamespace() == null || "tns".equals(identifier.getNamespace())) {
+            final String[] peekedWithNamespace = flowCallStack.peek().getFlowName().split("\\:");
+            String peekedNamespace = peekedWithNamespace[0];
+
+            flowCallStack.push(new FlowStackElement(peekedNamespace + ":" + identifier.getName(), null));
+          } else {
+            flowCallStack.push(new FlowStackElement(identifier.getNamespace() + ":" + identifier.getName(), null));
+          }
```

Comment:
Extract this, or just the `FlowStackElement` creation part to make the chain more readable

Reply:
Maybe just the `FlowStackElement`element creations, so its consistent with the pop below